src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -303,15 +304,23 @@ void WorldSession::HandleCharEnum(PreparedQueryResult result)
                 sWorld->AddCharacterInfo(charInfo.Guid, GetAccountId(), charInfo.Name, charInfo.Sex, charInfo.Race, charInfo.Class, charInfo.Level, false);
 
             if (charInfo.Class == CLASS_DEMON_HUNTER)
+                demonHunterCount++;
+
+            if (demonHunterCount >= sWorld->getIntConfig(CONFIG_DEMON_HUNTERS_PER_REALM) && !canAlwaysCreateDemonHunter)
                 charEnum.HasDemonHunterOnRealm = true;
-            if (charInfo.Level >= 70)
+            else
+                charEnum.HasDemonHunterOnRealm = false;
+
+            if (charInfo.Level >= sWorld->getIntConfig(CONFIG_CHARACTER_CREATING_MIN_LEVEL_FOR_DEMON_HUNTER))
                 charEnum.HasLevel70OnRealm = true;
 
             charEnum.Characters.emplace_back(charInfo);
         }
         while (result->NextRow());
     }
 
+    charEnum.IsDemonHunterCreationAllowed = (charEnum.HasDemonHunterOnRealm || !charEnum.HasLevel70OnRealm) ? false : true;
```

Comment:
the client needs the iscreation allowed as well as false else the disbaled button will not show the reason why you cannot create the demon hunter (eg. no lvl 70 character). tested ingame befor did that change